tests/test_task_engine_cstro.py

```diff
@@ -0,0 +1,1723 @@
+from prefect.filesystems import LocalFileSystem
+from prefect.logging import get_run_logger
+from prefect.results import PersistedResult, ResultFactory, UnpersistedResult
+from prefect.settings import (
+    PREFECT_EXPERIMENTAL_ENABLE_CLIENT_SIDE_TASK_ORCHESTRATION,
+    PREFECT_TASK_DEFAULT_RETRIES,
+    temporary_settings,
+)
+from prefect.states import Running, State
+from prefect.task_engine import TaskRunEngine, run_task_async, run_task_sync
+from prefect.task_runners import ThreadPoolTaskRunner
+from prefect.testing.utilities import exceptions_equal
+from prefect.utilities.callables import get_call_parameters
+from prefect.utilities.engine import propose_state
+
+
+@pytest.fixture(autouse=True)
+def enable_client_side_task_run_orchestration():
+    with temporary_settings(
+        {PREFECT_EXPERIMENTAL_ENABLE_CLIENT_SIDE_TASK_ORCHESTRATION: True}
+    ):
+        yield
+
+
+async def task_run_from_last_event(
+    asserting_events_worker: EventsWorker, task_run_id: Optional[UUID] = None
+) -> TaskRun:
+    events = sorted(asserting_events_worker._client.events, key=lambda e: e.occurred)
+    events = [e for e in events if e.event.startswith("prefect.task-run")]
+
+    if task_run_id:
+        events = [
+            e
+            for e in events
+            if e.resource.prefect_object_id("prefect.task-run") == task_run_id
+        ]
+
+    last_event = events[-1]
+    state = State(**last_event.payload["validated_state"])
+    return TaskRun(
+        id=last_event.resource.prefect_object_id("prefect.task-run"),
+        state=state,
+        state_id=state.id,
+        state_type=state.type,
+        state_name=state.name,
+        **last_event.payload["task_run"],
+    )
+
+
+@task
+async def foo():
+    return 42
+
+
+class TestTaskRunEngine:
+    async def test_basic_init(self):
+        engine = TaskRunEngine(task=foo)
+        assert isinstance(engine.task, Task)
+        assert engine.task.name == "foo"
+        assert engine.parameters == {}
+
+    async def test_client_attribute_raises_informative_error(self):
+        engine = TaskRunEngine(task=foo)
+        with pytest.raises(RuntimeError, match="not started"):
+            engine.client
+
+    async def test_client_attr_returns_client_after_starting(self):
+        engine = TaskRunEngine(task=foo)
+        with engine.initialize_run():
+            client = engine.client
+            assert isinstance(client, SyncPrefectClient)
+
+        with pytest.raises(RuntimeError, match="not started"):
+            engine.client
+
+
+class TestRunTask:
+    async def test_run_task_with_client_provided_uuid(
+        self,
+        asserting_events_worker: EventsWorker,
+    ):
+        @task
+        def foo():
+            return 42
+
+        task_run_id = uuid4()
+
+        run_task_sync(foo, task_run_id=task_run_id)
+
+        tr = await task_run_from_last_event(asserting_events_worker, task_run_id)
+        assert tr.id == task_run_id
```

Comment:
This entire file is a duplicate of `test_task_run_engine.py` with the flag enable and then changes to not look to the api for the task run. Example of what this test looked like before because there is no diff:
```
    def test_run_task_with_client_provided_uuid(
        self, sync_prefect_client: SyncPrefectClient
    ):
        @task
        def foo():
            return 42

        task_run_id = uuid4()

        run_task_sync(foo, task_run_id=task_run_id)

        task_run = sync_prefect_client.read_task_run(task_run_id)
        assert task_run.id == task_run_id
```